### PR TITLE
net: lib: ptp: Fix issues raised by static analyzer

### LINKED ISSUE #74799: [Coverity CID: 392505] Unused value in subsys/net/lib/ptp/transport.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c

Category: Incorrect expression
Function: `transport_udp_ipv4_open`
Component: Networking
CID: [392505](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392505)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c#L129

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392505

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_UNUSED_VALUE). ([CWE-563](http://cwe.mitre.org/data/definitions/563.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74800: [Coverity CID: 392508] Dereference before null check in subsys/net/lib/ptp/msg.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/msg.c

Category: Null pointer dereferences
Function: `ptp_msg_from_pkt`
Component: Networking
CID: [392508](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392508)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/msg.c#L289

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392508

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_REVERSE_INULL). ([CWE-476](http://cwe.mitre.org/data/definitions/476.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74801: [Coverity CID: 392511] Unchecked return value in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c

Category: Error handling issues
Function: `ptp_port_add_foreign_tt`
Component: Networking
CID: [392511](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392511)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L1363

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392511

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74804: [Coverity CID: 392523] Unsigned compared against 0 in subsys/net/lib/ptp/tlv.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/tlv.c#L82

Category: Control flow issues
Function: `tlv_mgmt_post_recv`
Component: Networking
CID: [392523](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392523)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/tlv.c#L82

```
76               data = mgmt_tlv->data;
77               data_length = length;
78    
79               clock_desc->type = (uint16_t *)data;
80               data += sizeof(*clock_desc->type);
81               data_length -= sizeof(*clock_desc->type);
>>>     CID 392523:  Control flow issues  (NO_EFFECT)
>>>     This less-than-zero comparison of an unsigned value is never true. "data_length < 0".
82               if (data_length < 0) {
83                       return -EBADMSG;
84               }
85               tlv_ntohs(&clock_desc->type);
86    
87               clock_desc->phy_protocol = (struct ptp_text *)data;
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_NO_EFFECT). ([CWE-570](http://cwe.mitre.org/data/definitions/570.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392523

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74805: [Coverity CID: 392526] Dereference null return value in subsys/net/lib/ptp/transport.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c#L327

Category: Null pointer dereferences
Function: `ptp_transport_protocol_addr`
Component: Networking
CID: [392526](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392526)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c#L327

```
321      int length = 0;
322    
323      if (IS_ENABLED(CONFIG_PTP_UDP_IPv4_PROTOCOL)) {
324              struct in_addr *ip = net_if_ipv4_get_global_addr(port->iface, NET_ADDR_PREFERRED);
325    
326              length = NET_IPV4_ADDR_SIZE;
>>>     CID 392526:  Null pointer dereferences  (NULL_RETURNS)
>>>     Dereferencing "ip", which is known to be "NULL".
327              *addr = ip->s_addr;
328      } else if (IS_ENABLED(CONFIG_PTP_UDP_IPv6_PROTOCOL)) {
329              struct in6_addr *ip = net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &port->iface);
330    
331              length = NET_IPV6_ADDR_SIZE;
332              memcpy(addr, ip, length);
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_NULL_RETURNS). ([CWE-476](http://cwe.mitre.org/data/definitions/476.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392526

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74806: [Coverity CID: 392530] Unintentional integer overflow in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L91

Category: Integer handling issues
Function: `port_timer_set_timeout`
Component: Networking
CID: [392530](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392530)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c

```
86       return ptp_transport_send(port, msg, idx);
87     }
88    
89     static void port_timer_set_timeout(struct k_timer *timer, uint8_t factor, int8_t log_seconds)
90     {
91       uint64_t timeout = log_seconds < 0 ? (NSEC_PER_SEC * factor) >> (log_seconds * -1) :
>>>     CID 392530:    (OVERFLOW_BEFORE_WIDEN)
>>>     Potentially overflowing expression "1000000000U * factor" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "uint64_t" (64 bits, unsigned).
92                                            (NSEC_PER_SEC * factor) << log_seconds;
93    
94       k_timer_start(timer, K_NSEC(timeout), K_NO_WAIT);
95     }
96    
97     static void port_timer_set_timeout_random(struct k_timer *timer,
85    
86       return ptp_transport_send(port, msg, idx);
87     }
88    
89     static void port_timer_set_timeout(struct k_timer *timer, uint8_t factor, int8_t log_seconds)
90     {
>>>     CID 392530:    (OVERFLOW_BEFORE_WIDEN)
>>>     Potentially overflowing expression "1000000000U * factor" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "uint64_t" (64 bits, unsigned).
91       uint64_t timeout = log_seconds < 0 ? (NSEC_PER_SEC * factor) >> (log_seconds * -1) :
92                                            (NSEC_PER_SEC * factor) << log_seconds;
93    
94       k_timer_start(timer, K_NSEC(timeout), K_NO_WAIT);
95     }
96    
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERFLOW_BEFORE_WIDEN). ([CWE-190](http://cwe.mitre.org/data/definitions/190.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392530

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74807: [Coverity CID: 392532] Unintentional integer overflow in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L109

Category: Integer handling issues
Function: `port_timer_set_timeout_random`
Component: Networking
CID: [392532](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392532)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L109

```
103    
104      if (log_seconds < 0) {
105              timeout = (NSEC_PER_SEC * min_factor) >> -log_seconds;
106              random_ns = NSEC_PER_SEC >> -log_seconds;
107      } else {
108              timeout = (NSEC_PER_SEC * min_factor) << log_seconds;
>>>     CID 392532:  Integer handling issues  (OVERFLOW_BEFORE_WIDEN)
>>>     Potentially overflowing expression "span * 1000000000U" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "uint64_t" (64 bits, unsigned).
109              random_ns = (span * NSEC_PER_SEC) << log_seconds;
110      }
111    
112      timeout = timeout + (random_ns * (sys_rand32_get() % (1 << 15) + 1) >> 15);
113      k_timer_start(timer, K_NSEC(timeout), K_NO_WAIT);
114     }
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERFLOW_BEFORE_WIDEN). ([CWE-190](http://cwe.mitre.org/data/definitions/190.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392532

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74808: [Coverity CID: 392534] Unused value in subsys/net/lib/ptp/transport.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c#L177

Category: Incorrect expression
Function: `transport_udp_ipv6_open`
Component: Networking
CID: [392534](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392534)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/transport.c#L177

```
171      }
172    
173      if (zsock_getsockopt(socket, IPPROTO_IPV6, IPV6_TCLASS, &tclass, &length)) {
174              tclass = 0;
175      }
176    
>>>     CID 392534:  Incorrect expression  (UNUSED_VALUE)
>>>     Assigning value "-253" to "tclass" here, but that stored value is overwritten before it can be used.
177      tclass &= ~0xFC;
178      tclass = CONFIG_PTP_DSCP_VALUE << 2;
179      length = sizeof(tclass);
180    
181      if (zsock_setsockopt(socket, IPPROTO_IPV6, IPV6_TCLASS, &tclass, length)) {
182              LOG_WRN("Failed to set priority");
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_UNUSED_VALUE). ([CWE-563](http://cwe.mitre.org/data/definitions/563.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392534

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74809: [Coverity CID: 392535] Out-of-bounds write in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L1019

Category: Memory - corruptions
Function: `ptp_port_init`
Component: Networking
CID: [392535](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392535)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L1019

```
1013                     LOG_WRN("Exceeded number of PTP Ports.");
1014                     return;
1015             }
1016    
1017             port = ports + dds->n_ports;
1018    
>>>     CID 392535:  Memory - corruptions  (OVERRUN)
>>>     Overrunning array of 364 bytes at byte offset 364 by dereferencing pointer "port".
1019             port->iface = iface;
1020             port->best = NULL;
1021             port->socket[PTP_SOCKET_EVENT] = -1;
1022             port->socket[PTP_SOCKET_GENERAL] = -1;
1023    
1024             port->state_machine = dds->time_receiver_only ? ptp_tr_state_machine : ptp_state_machine;
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERRUN). ([CWE-119](http://cwe.mitre.org/data/definitions/119.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392535

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74811: [Coverity CID: 392538] Unintentional integer overflow in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L105

Category: Integer handling issues
Function: `port_timer_set_timeout_random`
Component: Networking
CID: [392538](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392538)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c

```
102      uint64_t timeout, random_ns;
103    
104      if (log_seconds < 0) {
105              timeout = (NSEC_PER_SEC * min_factor) >> -log_seconds;
106              random_ns = NSEC_PER_SEC >> -log_seconds;
107      } else {
>>>     CID 392538:    (OVERFLOW_BEFORE_WIDEN)
>>>     Potentially overflowing expression "1000000000U * min_factor" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "uint64_t" (64 bits, unsigned).
108              timeout = (NSEC_PER_SEC * min_factor) << log_seconds;
109              random_ns = (span * NSEC_PER_SEC) << log_seconds;
110      }
111    
112      timeout = timeout + (random_ns * (sys_rand32_get() % (1 << 15) + 1) >> 15);
113      k_timer_start(timer, K_NSEC(timeout), K_NO_WAIT);
99                                         int span,
100                                        int log_seconds)
101     {
102      uint64_t timeout, random_ns;
103    
104      if (log_seconds < 0) {
>>>     CID 392538:    (OVERFLOW_BEFORE_WIDEN)
>>>     Potentially overflowing expression "1000000000U * min_factor" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "uint64_t" (64 bits, unsigned).
105              timeout = (NSEC_PER_SEC * min_factor) >> -log_seconds;
106              random_ns = NSEC_PER_SEC >> -log_seconds;
107      } else {
108              timeout = (NSEC_PER_SEC * min_factor) << log_seconds;
109              random_ns = (span * NSEC_PER_SEC) << log_seconds;
110      }
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERFLOW_BEFORE_WIDEN). ([CWE-190](http://cwe.mitre.org/data/definitions/190.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392538

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #74812: [Coverity CID: 392541] Self assignment in subsys/net/lib/ptp/port.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L980

Category: Incorrect expression
Function: `port_link_monitor`
Component: Networking
CID: [392541](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392541)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/net/lib/ptp/port.c#L980

```
974    
975      if (!port) {
976              return;
977      }
978    
979      if (iface_state & port->link_status) {
>>>     CID 392541:  Incorrect expression  (NO_EFFECT)
>>>     Assigning "port->link_status" to itself has no effect.
980              port->link_status = port->link_status;
981      } else {
982              port->link_status = iface_state | PORT_LINK_CHANGED;
983              LOG_DBG("Port %d link %s",
984                      port->port_ds.id.port_number,
985                      port->link_status & PORT_LINK_UP ? "up" : "down");
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_NO_EFFECT). ([CWE-665](http://cwe.mitre.org/data/definitions/665.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392541

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by awojasinski**
Fix minor issues reported by static analysis tool:
- #74799
- #74800 
- #74801
- #74804 
- #74805 
- #74806 
- #74807 
- #74808 
- #74809
- #74811 
- #74812

Fixes #74799
Fixes #74800 
Fixes #74801
Fixes #74804 
Fixes #74805 
Fixes #74806 
Fixes #74807 
Fixes #74808 
Fixes #74809
Fixes #74811 
Fixes #74812

### CODE REVIEW

File: subsys/net/lib/ptp/msg.c
Comment (awojasinski) [inline subsys/net/lib/ptp/msg.c:1]: Fixes #74800

File: subsys/net/lib/ptp/port.c [hunk 1]
```diff
@@ -88,8 +88,9 @@ static int port_msg_send(struct ptp_port *port, struct ptp_msg *msg, enum ptp_so
 
 static void port_timer_set_timeout(struct k_timer *timer, uint8_t factor, int8_t log_seconds)
 {
-	uint64_t timeout = log_seconds < 0 ? (NSEC_PER_SEC * factor) >> (log_seconds * -1) :
-					     (NSEC_PER_SEC * factor) << log_seconds;
+	uint64_t timeout = log_seconds < 0 ?
+		(uint64_t)((NSEC_PER_SEC * factor) >> (log_seconds * -1)) :
+		(uint64_t)((NSEC_PER_SEC * factor) << log_seconds);
```
Comment (awojasinski) [inline subsys/net/lib/ptp/port.c:93]: Fixes #74806

File: subsys/net/lib/ptp/port.c [hunk 2]
```diff
@@ -102,14 +103,14 @@ static void port_timer_set_timeout_random(struct k_timer *timer,
...
+		random_ns = (uint64_t)(NSEC_PER_SEC >> -log_seconds);
 	} else {
-		timeout = (NSEC_PER_SEC * min_factor) << log_seconds;
-		random_ns = (span * NSEC_PER_SEC) << log_seconds;
+		timeout = (uint64_t)((NSEC_PER_SEC * min_factor) << log_seconds);
+		random_ns = (uint64_t)((span * NSEC_PER_SEC) << log_seconds);
 	}
 
-	timeout = timeout + (random_ns * (sys_rand32_get() % (1 << 15) + 1) >> 15);
+	timeout = (uint64_t)(timeout + (random_ns * (sys_rand32_get() % (1 << 15) + 1) >> 15));
```
Comment (awojasinski) [inline subsys/net/lib/ptp/port.c:113]: Fixes #74807

File: subsys/net/lib/ptp/port.c [hunk 3]
```diff
@@ -970,14 +971,14 @@ static void port_link_monitor(struct net_mgmt_event_callback *cb,
...
-	int iface_state = mgmt_event == NET_EVENT_IF_UP ? PORT_LINK_UP : PORT_LINK_DOWN;
+	uint8_t iface_state = mgmt_event == NET_EVENT_IF_UP ? PORT_LINK_UP : PORT_LINK_DOWN;
 
 	if (!port) {
 		return;
 	}
 
 	if (iface_state & port->link_status) {
-		port->link_status = port->link_status;
+		port->link_status = iface_state;
```
Comment (awojasinski) [inline subsys/net/lib/ptp/port.c:981]: Fixes #74812

File: subsys/net/lib/ptp/port.c [hunk 4]
```diff
@@ -1014,7 +1015,7 @@ void ptp_port_init(struct net_if *iface, void *user_data)
 		return;
 	}
 
-	port = ports + dds->n_ports;
+	port = &ports[dds->n_ports];
```
Comment (awojasinski) [inline subsys/net/lib/ptp/port.c:1018]: Fixes #74809

File: subsys/net/lib/ptp/port.c [hunk 5]
```diff
@@ -1360,7 +1361,7 @@ int ptp_port_add_foreign_tt(struct ptp_port *port, struct ptp_msg *msg)
 			port->port_ds.id.port_number,
 			port_id_str(&msg->header.src_port_id));
 
-		k_mem_slab_alloc(&foreign_tts_slab, (void **)&foreign, K_NO_WAIT);
+		foreign = k_mem_slab_alloc(&foreign_tts_slab, (void **)&foreign, K_NO_WAIT);
```
Comment (awojasinski) [inline subsys/net/lib/ptp/port.c:1364]: Fixes #74801

File: subsys/net/lib/ptp/tlv.c [hunk 6]
```diff
@@ -51,7 +51,7 @@ static int tlv_mgmt_post_recv(struct ptp_tlv_mgmt *mgmt_tlv, uint16_t length)
 	enum ptp_mgmt_id id = (enum ptp_mgmt_id)mgmt_tlv->id;
 	struct ptp_tlv_container *container;
 	uint8_t *data;
-	uint16_t data_length;
+	int32_t data_length;
```
Comment (awojasinski) [inline subsys/net/lib/ptp/tlv.c:54]: Fixes #74804

File: subsys/net/lib/ptp/transport.c [hunk 7]
```diff
@@ -127,7 +127,7 @@ static int transport_udp_ipv4_open(struct net_if *iface, uint16_t port)
 	}
 
 	tos &= ~0xFC;
-	tos = CONFIG_PTP_DSCP_VALUE << 2;
+	tos |= CONFIG_PTP_DSCP_VALUE << 2;
```
Comment (awojasinski) [inline subsys/net/lib/ptp/transport.c:130]: Fixes #74799

File: subsys/net/lib/ptp/transport.c [hunk 8]
```diff
@@ -175,7 +175,7 @@ static int transport_udp_ipv6_open(struct net_if *iface, uint16_t port)
 	}
 
 	tclass &= ~0xFC;
-	tclass = CONFIG_PTP_DSCP_VALUE << 2;
+	tclass |= CONFIG_PTP_DSCP_VALUE << 2;
```
Comment (awojasinski) [inline subsys/net/lib/ptp/transport.c:178]: Fixes #74808

File: subsys/net/lib/ptp/transport.c [hunk 9]
```diff
@@ -318,6 +318,8 @@ int ptp_transport_recv(struct ptp_port *port, struct ptp_msg *msg, enum ptp_sock
 
 int ptp_transport_protocol_addr(struct ptp_port *port, uint8_t *addr)
 {
+	__ASSERT_NO_MSG(addr);
```
Comment (awojasinski) [inline subsys/net/lib/ptp/transport.c:321]: Fixes #74805

**Comment by awojasinski** [comment]
typo in commit message

**Comment by jukkar** [comment]
@awojasinski do you have access to coverity site? It would be great if these issues could be marked as resolved or false positive there so that coverity will not report these issues again.

**Comment by awojasinski** [comment]
@jukkar I don't have access. Can you grant me an access or should I ask someone else?

**Comment by nashif** [comment]
> @awojasinski do you have access to coverity site? It would be great if these issues could be marked as resolved or false positive there so that coverity will not report these issues again.

coverity will not report those again if the fix is correct for a real issue, for false positive, yes, they need to be marked.

**Comment by jukkar** [comment]
I cannot create account there, perhaps @nashif can guide you here.

**Comment by jukkar** [comment]
If all the issues were real, so no false positives, then I think we are fine and nothing needs to be set in coverity site.

**Comment by awojasinski** [comment]
> If all the issues were real, so no false positives, then I think we are fine and nothing needs to be set in coverity site.

there is one false positive I've already closed #74803. @nashif if you could mark it as false positive in coverity I would be very grateful.

**Comment by jukkar** [comment]
> there is one false positive I've already closed #74803. @nashif if you could mark it as false positive in coverity I would be very grateful.

No need to bother Anas with such a minor detail, I have an account in Coverity so I can fix that tomorrow. But as you are the maintainer there might be new coverity issues in the future so it would make sense to create an account there, it just does not scale if someone else needs to tweak coverity for those issues. I am not sure how the account is created, I did that so many years ago that I have forgotten such a minor detail. Anyway, Anas might be able to help you to create the account.

**Comment by jukkar** [comment]
The one false positive is now marked as such in Coverity.

**Comment by awojasinski** [comment]
thanks @jukkar

**Review by jukkar — APPROVED** [review]
LGTM. Usually we have one Coverity fix / commit, but I can certainly understand motivation to bundle multiple fixes to one commit. It is just that now it is more difficult to know what fix is related to what issue, without looking the coverity report.

**Comment by awojasinski** [comment]
I could split it into separate commits if you would like but I thought there is no need to have several one line commits

**Review by rlubos — APPROVED** [review]

**Comment by jukkar** [comment]
> I could split it into separate commits if you would like but I thought there is no need to have several one line commits

No worries, this is ok too.

**Review by srv-meta — APPROVED** [review]

**Pull request merged**